monai/apps/deepgrow/dataset.py

```diff
@@ -0,0 +1,271 @@
+    mode = [GridSampleMode.BILINEAR, GridSampleMode.NEAREST] if len(keys) == 2 else [GridSampleMode.BILINEAR]
+    transforms = [
+        LoadImaged(keys=keys),
+        AsChannelFirstd(keys=keys),
+        Spacingd(keys=keys, pixdim=pixdim, mode=mode),
+        Orientationd(keys=keys, axcodes="RAS"),
+    ]
+
+    return Compose(transforms)
+
+
+def _save_data_2d(vol_idx, data, keys, dataset_dir, relative_path):
+    vol_image = data[keys[0]]
+    vol_label = data.get(keys[1])
+    data_list = []
+
+    if len(vol_image.shape) == 4:
+        logging.info("4D-Image, pick only first series; Image: {}; Label: {}".format(vol_image.shape, vol_label.shape))
+        vol_image = vol_image[0]
+        vol_image = np.moveaxis(vol_image, -1, 0)
+
+    image_count = 0
+    label_count = 0
+    unique_labels_count = 0
+    for sid in range(vol_image.shape[0]):
+        image = vol_image[sid, ...]
+        label = vol_label[sid, ...] if vol_label is not None else None
+
+        if vol_label is not None and np.sum(label) == 0:
+            continue
+
+        image_file_prefix = "vol_idx_{:0>4d}_slice_{:0>3d}".format(vol_idx, sid)
+        image_file = os.path.join(dataset_dir, "images", image_file_prefix)
+        image_file += ".npy"
+
+        os.makedirs(os.path.join(dataset_dir, "images"), exist_ok=True)
+        np.save(image_file, image)
+        image_count += 1
+
+        # Test Data
+        if vol_label is None:
+            data_list.append(
+                {
+                    "image": image_file.replace(dataset_dir + "/", "") if relative_path else image_file,
+                }
+            )
+            continue
+
+        # For all Labels
+        unique_labels = np.unique(label.flatten())
+        unique_labels = unique_labels[unique_labels != 0]
+        unique_labels_count = max(unique_labels_count, len(unique_labels))
+
+        for idx in unique_labels:
+            label_file_prefix = "{}_region_{:0>2d}".format(image_file_prefix, int(idx))
+            label_file = os.path.join(dataset_dir, "labels", label_file_prefix)
+            label_file += ".npy"
+
+            os.makedirs(os.path.join(dataset_dir, "labels"), exist_ok=True)
+            curr_label = (label == idx).astype(np.float32)
+            np.save(label_file, curr_label)
+
+            label_count += 1
+            data_list.append(
+                {
+                    "image": image_file.replace(dataset_dir + "/", "") if relative_path else image_file,
+                    "label": label_file.replace(dataset_dir + "/", "") if relative_path else label_file,
+                    "region": int(idx),
+                }
+            )
+
+    print(
+        "{} => Image: {} => {}; Label: {} => {}; Unique Labels: {}".format(
+            vol_idx,
+            vol_image.shape,
+            image_count,
+            vol_label.shape if vol_label is not None else None,
+            label_count,
+            unique_labels_count,
+        )
+    )
```

Comment:
these prints need to be either "more descriptive" or we just remove them?
Or we put them in some kind of debug messages.
Because 1 => Image: 1,128,128,128 => 8; Label: 1,128,128,128 => 8; Unique Labels: 10
is not very clear to the users?

Reply:
ixed